### PR TITLE
Empty Directories, Hide files, ignoreList

### PR DESCRIPTION
**Pull request by Meistercoach83**
Added possibilty for 
- showing/hiding files
- showing/hiding empty directories
- ignoreList

### CODE REVIEW

**Comment by Meistercoach83** [comment]
okay - now my tests are working. but i removed the "should display the size of a directory (summing up the children)" test.

File: lib/directory-tree.js [hunk 1]
```diff
@@ -1,54 +1,65 @@
...
+        };
+
+        var stats = fs.statSync(name);
 
         var item = {
             path: path.relative(basepath, name),
             name: path.basename(name)
         };
 
+        if (stats.isFile() && options.hideFiles) {
```
Comment (mihneadb) [inline lib/directory-tree.js]: Please invert the order of the two here. No need to check `isFile` if `hideFiles` is true.

File: lib/directory-tree.js [hunk 2]
```diff
@@ -1,54 +1,65 @@
...
+
+        if (options.ignoreList.length > 0) { 
+            var ignore = false;
+            options.ignoreList.forEach(function (value) {
+                if (name.indexOf(value) > -1) { 
+                    ignore = true;
+                    return null;
+                }
+            });
+            if(ignore){
```
Comment (mihneadb) [inline lib/directory-tree.js]: Whitespace.

File: lib/directory-tree.js [hunk 3]
```diff
@@ -1,54 +1,65 @@
...
+        if (stats.isFile() && options.hideFiles) {
+            return null;
+        }
+
+        if (options.ignoreList.length > 0) { 
+            var ignore = false;
+            options.ignoreList.forEach(function (value) {
+                if (name.indexOf(value) > -1) { 
+                    ignore = true;
+                    return null;
```
Comment (mihneadb) [inline lib/directory-tree.js]: You don't really need this return statement, forEach will execute on every element anyway.

File: lib/directory-tree.js [hunk 4]
```diff
@@ -1,54 +1,65 @@
...
+                return null;
+            }
+        }
+
         if (stats.isFile()) {
-            if (extensions &&
-                extensions.length > 0 &&
-                extensions.indexOf(path.extname(name).toLowerCase()) == -1 &&
-                extensions.indexOf(path.basename(name)) > -1) {
+            if (options.fileExtensions.length > 0 && options.fileExtensions.indexOf(path.extname(name).toLowerCase()) === -1) {
```
Comment (mihneadb) [inline lib/directory-tree.js]: Please keep this on multiple lines, as it was before.

File: lib/directory-tree.js [hunk 5]
```diff
@@ -1,54 +1,65 @@
...
-                return _directoryTree(path.join(name, child), extensions);
+                return _directoryTree(path.join(name, child), options);
             }).filter(function (e) {
-                return e != null;
+                return e !== null;
             });
-
-            if (item.children.length == 0) {
+            
+            if (item.children.length === 0 && options.hideEmptyDirectories) {
```
Comment (mihneadb) [inline lib/directory-tree.js]: Please invert the two, same reason as before (for `isFile()`).

File: lib/directory-tree.js [hunk 6]
```diff
@@ -1,54 +1,65 @@
...
             item.children = fs.readdirSync(name).map(function (child) {
-                return _directoryTree(path.join(name, child), extensions);
+                return _directoryTree(path.join(name, child), options);
             }).filter(function (e) {
-                return e != null;
+                return e !== null;
             });
-
-            if (item.children.length == 0) {
+
```
Comment (mihneadb) [inline lib/directory-tree.js]: Trailing whitespace.

File: test/test.js [hunk 7]
```diff
@@ -1,20 +1,88 @@
-var expect = require('chai').expect;
-var dirtree = require('../lib/directory-tree');
+var expect = require('chai').expect,
+        dirtree = require('../lib/directory-tree'),
```
Comment (mihneadb) [inline test/test.js]: You probably tried to align these, but it didn't work out. I suggest you keep the old format.

File: test/test.js [hunk 8]
```diff
@@ -1,20 +1,88 @@
...
+        fs = require('fs');
 
+describe('directoryTree', function () {
 
-describe('directoryTree', function() {
-  it('should return an Object', function() {
-    var tree = dirtree.directoryTree('./test/test_data', ['.DS_Store', '.gitkeep']);
-    expect(tree).to.be.an('object');
-  });
+    it('should return an Object', function () {
```
Comment (mihneadb) [inline test/test.js]: You changed the file's indent rules, please stick to what was there before.

File: test/test.js [hunk 9]
```diff
@@ -1,20 +1,88 @@
...
+        it('should hide files in a directory listing', function () {
+            var tree = dirtree.directoryTree('./test/test_data', {'hideFiles': true});
+            expect(tree.children.length).to.equal(2);
+        });
+    });
+
+    describe('Tests for ignoring files and / or directories', function () {
+
+        it('should ignore files', function () {
+            var tree = dirtree.directoryTree('./test/test_data/some_dir_2', {'ignoreList': ['.DS_Store', '.gitkeep']});
```
Comment (mihneadb) [inline test/test.js]: Split on two lines pls.

**Comment by mihneadb** [comment]
Hi, thanks for your contribution!

1. why did you remove the size logic?
2. to be complete, you should've removed the size parts from the readme as well

File: test/test.js (hunk 7 above)
Comment (Meistercoach83) [inline test/test.js]: with this format it´s only shorter to define the vars.

File: test/test.js (hunk 8 above)
Comment (Meistercoach83) [inline test/test.js]: can you tell me the rules? i used the netbeans standard rules

**Comment by Meistercoach83** [comment]
I got an .size is undefined when i tried the directory test.

**Comment by mihneadb** [comment]
@Meistercoach83 well then I think you should rather fix the bug that led to that rather than removing the functionality :). (sorry for the delay!)

**Comment by Meistercoach83** [comment]
sorry - i wasn´t at home the last week. It´s on my todo list.

**Comment by Meistercoach83** [comment]
still open?

**Comment by mihneadb** [comment]
@Meistercoach83 I don't get notifications when someone pushes a commit. Only comments trigger notifications! Sorry for missing your commit. I'll check the PR again over the weekend!

File: test/test.js (hunk 8 above)
Comment (mihneadb) [inline test/test.js]: 2 spaces indent, pretty much.

**Comment by mihneadb** [comment]
@Meistercoach83 so pls just update the indent to 2 spaces and it should be good to go! Thanks!

**Comment by Meistercoach83** [comment]
@mihneadb ok?

File: package.json [hunk 10]
```diff
@@ -24,5 +26,12 @@
...
+  "readmeFilename": "README.md",
+  "_id": "directory-tree@0.1.1",
```
Comment (mihneadb) [inline package.json]: Not sure, are these fields below needed?

**Comment by mihneadb** [comment]
@Meistercoach83 looks good, ty. Please take a look at package json, I think some extra lines were added and they weren't supposed to.

**Comment by mihneadb** [comment]
This somehow got off the radar. Closing.

**Pull request closed without merging**